### PR TITLE
avoid conflict java.lang.Object and scala.Product methods

### LINKED ISSUE #671: columnNameToFieldName setting does not work correctly
scalikejdbc-mapper-generator-core/src/main/scala/scalikejdbc/mapper/GeneratorConfig.scala

```diff
-  columnNameToFieldName: String => String = GeneratorConfig.lowerCamelCase andThen GeneratorConfig.quoteReservedWord,
+  columnNameToFieldName: String => String = GeneratorConfig.lowerCamelCase andThen GeneratorConfig.quoteReservedWord andThen GeneratorConfig.addSuffixIfConflict("Column"),
   returnCollectionType: ReturnCollectionType = ReturnCollectionType.List,
   view: Boolean = false
 )
 @@ -47,6 +47,17 @@ object GeneratorConfig {
       else name
   }
 
+  val conflictMethods: Set[String] = Set(
+    "toString", "hashCode", "wait", "getClass", "notify", "notifyAll",
+    "productArity", "productIterator", "productPrefix"
+  )
+
+  def addSuffixIfConflict(suffix: String): String => String = {
+    name =>
+      if (conflictMethods(name)) name + suffix
+      else name
+  }
```

scalikejdbc-mapper-generator/src/sbt-test/scalikejdbc-mapper-generator/gen/build.sbt

```diff
   DB.autoCommit { implicit s =>
-    sql"create table if not exists programmers (id SERIAL PRIMARY KEY, name varchar(128), t1 timestamp not null, t2 date, t3 time, type int)"
+    sql"create table if not exists programmers (id SERIAL PRIMARY KEY, name varchar(128), t1 timestamp not null, t2 date, t3 time, type int, to_string int, hash_code int, wait int, get_class int, notify int, notify_all int, product_arity int, product_iterator int, product_prefix int)"
       .execute.apply()
     sql"create view programmers_view as (select * from programmers)"
       .execute.apply()
 @@ -37,7 +37,6 @@ val scalikejdbcVersion = System.getProperty("plugin.version")
 
 crossScalaVersions := List("2.12.1", "2.11.8", "2.10.6") 
```

```
[error] scalikejdbc.InvalidColumnNameException: Invalid column name. (name: to_string_column, registered names: id,name,t1,t2,t3,type,to_string,hash_code,wait,get_class,notify,notify_all,product_arity,product_iterator,product_prefix)
[error] 	at scalikejdbc.SQLSyntaxSupportFeature$ColumnSQLSyntaxProvider.$anonfun$column$4(SQLSyntaxSupportFeature.scala:451)
[error] 	at scalikejdbc.SQLSyntaxSupportFeature$ColumnSQLSyntaxProvider$$Lambda$1773/1973869753.apply(Unknown Source)
[error] 	at scala.Option.getOrElse(Option.scala:121)
[error] 	at scalikejdbc.SQLSyntaxSupportFeature$ColumnSQLSyntaxProvider.$anonfun$column$1(SQLSyntaxSupportFeature.scala:450)
[error] 	at scalikejdbc.SQLSyntaxSupportFeature$ColumnSQLSyntaxProvider$$Lambda$1768/1333090058.apply(Unknown Source)
[error] 	at scala.collection.concurrent.TrieMap.getOrElseUpdate(TrieMap.scala:903)
[error] 	at scalikejdbc.SQLSyntaxSupportFeature$ColumnSQLSyntaxProvider.column(SQLSyntaxSupportFeature.scala:449)
[error] 	at scalikejdbc.SQLSyntaxSupportFeature$SQLSyntaxProvider.c(SQLSyntaxSupportFeature.scala:309)
[error] 	at scalikejdbc.SQLSyntaxSupportFeature$SQLSyntaxProvider.c$(SQLSyntaxSupportFeature.scala:309)
[error] 	at scalikejdbc.SQLSyntaxSupportFeature$ColumnSQLSyntaxProvider.c(SQLSyntaxSupportFeature.scala:421)
[error] 	at scalikejdbc.SQLSyntaxSupportFeature$SQLSyntaxProvider.field(SQLSyntaxSupportFeature.scala:324)
[error] 	at scalikejdbc.SQLSyntaxSupportFeature$SQLSyntaxProvider.field$(SQLSyntaxSupportFeature.scala:319)
[error] 	at scalikejdbc.SQLSyntaxSupportFeature$ColumnSQLSyntaxProvider.field(SQLSyntaxSupportFeature.scala:421)
[error] 	at app.models.Programmers$.create(Programmers.scala:113)
[error] 	at app.models.ProgrammersTest.$anonfun$new$2(ProgrammersTest.scala:20)
[error] 	at app.models.ProgrammersTest$$Lambda$1760/1580591915.apply(Unknown Source)
[error] 	at scala.collection.TraversableLike.$anonfun$map$1(TraversableLike.scala:234)
[error] 	at scala.collection.TraversableLike$$Lambda$92/1083115915.apply(Unknown Source)
[error] 	at scala.collection.immutable.List.foreach(List.scala:378)
[error] 	at scala.collection.TraversableLike.map(TraversableLike.scala:234)
[error] 	at scala.collection.TraversableLike.map$(TraversableLike.scala:227)
[error] 	at scala.collection.immutable.List.map(List.scala:284)
[error] 	at app.models.ProgrammersTest.$anonfun$new$1(ProgrammersTest.scala:19)
[error] 	at app.models.ProgrammersTest$$Lambda$1511/1045217306.apply(Unknown Source)
[error] 	at org.scalatest.fixture.TestSuite$TestFunAndConfigMap.$anonfun$apply$1(TestSuite.scala:132)
[error] 	at org.scalatest.fixture.TestSuite$TestFunAndConfigMap$$Lambda$1566/1181138909.apply(Unknown Source)
[error] 	at org.scalatest.OutcomeOf.outcomeOf(OutcomeOf.scala:85)
[error] 	at org.scalatest.OutcomeOf.outcomeOf$(OutcomeOf.scala:83)
[error] 	at org.scalatest.OutcomeOf$.outcomeOf(OutcomeOf.scala:104)
[error] 	at org.scalatest.fixture.TestSuite$TestFunAndConfigMap.apply(TestSuite.scala:132)
[error] 	at org.scalatest.fixture.TestSuite$OneArgTest$$anon$1.apply(TestSuite.scala:80)
[error] 	at org.scalatest.TestSuite.withFixture(TestSuite.scala:196)
[error] 	at org.scalatest.TestSuite.withFixture$(TestSuite.scala:195)
[error] 	at org.scalatest.fixture.FlatSpec.withFixture(FlatSpec.scala:226)
[error] 	at scalikejdbc.scalatest.AutoRollback.$anonfun$withFixture$1(AutoRollback.scala:63)
[error] 	at scalikejdbc.scalatest.AutoRollback$$Lambda$1564/1342568117.apply(Unknown Source)
[error] 	at scalikejdbc.LoanPattern.using(LoanPattern.scala:18)
[error] 	at scalikejdbc.LoanPattern.using$(LoanPattern.scala:16)
[error] 	at app.models.ProgrammersTest.using(ProgrammersTest.scala:9)
[error] 	at scalikejdbc.scalatest.AutoRollback.withFixture(AutoRollback.scala:57)
[error] 	at scalikejdbc.scalatest.AutoRollback.withFixture$(AutoRollback.scala:56)
[error] 	at app.models.ProgrammersTest.withFixture(ProgrammersTest.scala:9)
[error] 	at org.scalatest.fixture.FlatSpecLike.invokeWithFixture$1(FlatSpecLike.scala:2131)
[error] 	at org.scalatest.fixture.FlatSpecLike.$anonfun$runTest$1(FlatSpecLike.scala:2142)
[error] 	at org.scalatest.fixture.FlatSpecLike$$Lambda$1551/2097967795.apply(Unknown Source)
[error] 	at org.scalatest.SuperEngine.runTestImpl(Engine.scala:289)
[error] 	at org.scalatest.fixture.FlatSpecLike.runTest(FlatSpecLike.scala:2142)
[error] 	at org.scalatest.fixture.FlatSpecLike.runTest$(FlatSpecLike.scala:2123)
[error] 	at org.scalatest.fixture.FlatSpec.runTest(FlatSpec.scala:226)
[error] 	at org.scalatest.fixture.FlatSpecLike.$anonfun$runTests$1(FlatSpecLike.scala:2185)
[error] 	at org.scalatest.fixture.FlatSpecLike$$Lambda$1535/1738234703.apply(Unknown Source)
[error] 	at org.scalatest.SuperEngine.$anonfun$runTestsInBranch$1(Engine.scala:396)
[error] 	at org.scalatest.SuperEngine$$Lambda$1536/1519856649.apply(Unknown Source)
[error] 	at scala.collection.immutable.List.foreach(List.scala:378)
[error] 	at org.scalatest.SuperEngine.traverseSubNodes$1(Engine.scala:384)
[error] 	at org.scalatest.SuperEngine.runTestsInBranch(Engine.scala:373)
[error] 	at org.scalatest.SuperEngine.$anonfun$runTestsInBranch$1(Engine.scala:410)
[error] 	at org.scalatest.SuperEngine$$Lambda$1536/1519856649.apply(Unknown Source)
[error] 	at scala.collection.immutable.List.foreach(List.scala:378)
[error] 	at org.scalatest.SuperEngine.traverseSubNodes$1(Engine.scala:384)
[error] 	at org.scalatest.SuperEngine.runTestsInBranch(Engine.scala:379)
[error] 	at org.scalatest.SuperEngine.runTestsImpl(Engine.scala:461)
[error] 	at org.scalatest.fixture.FlatSpecLike.runTests(FlatSpecLike.scala:2185)
[error] 	at org.scalatest.fixture.FlatSpecLike.runTests$(FlatSpecLike.scala:2184)
[error] 	at org.scalatest.fixture.FlatSpec.runTests(FlatSpec.scala:226)
[error] 	at org.scalatest.Suite.run(Suite.scala:1147)
[error] 	at org.scalatest.Suite.run$(Suite.scala:1129)
[error] 	at org.scalatest.fixture.FlatSpec.org$scalatest$fixture$FlatSpecLike$$super$run(FlatSpec.scala:226)
[error] 	at org.scalatest.fixture.FlatSpecLike.$anonfun$run$1(FlatSpecLike.scala:2206)
[error] 	at org.scalatest.fixture.FlatSpecLike$$Lambda$1528/1346345487.apply(Unknown Source)
[error] 	at org.scalatest.SuperEngine.runImpl(Engine.scala:521)
[error] 	at org.scalatest.fixture.FlatSpecLike.run(FlatSpecLike.scala:2206)
[error] 	at org.scalatest.fixture.FlatSpecLike.run$(FlatSpecLike.scala:2205)
[error] 	at org.scalatest.fixture.FlatSpec.run(FlatSpec.scala:226)
[error] 	at org.scalatest.tools.Framework.org$scalatest$tools$Framework$$runSuite(Framework.scala:314)
[error] 	at org.scalatest.tools.Framework$ScalaTestTask.execute(Framework.scala:507)
[error] 	at sbt.TestRunner.runTest$1(TestFramework.scala:84)
[error] 	at sbt.TestRunner.run(TestFramework.scala:94)
[error] 	at sbt.TestFramework$$anon$2$$anonfun$$init$$1$$anonfun$apply$8.apply(TestFramework.scala:219)
[error] 	at sbt.TestFramework$$anon$2$$anonfun$$init$$1$$anonfun$apply$8.apply(TestFramework.scala:219)
[error] 	at sbt.TestFramework$.sbt$TestFramework$$withContextLoader(TestFramework.scala:207)
[error] 	at sbt.TestFramework$$anon$2$$anonfun$$init$$1.apply(TestFramework.scala:219)
[error] 	at sbt.TestFramework$$anon$2$$anonfun$$init$$1.apply(TestFramework.scala:219)
[error] 	at sbt.TestFunction.apply(TestFramework.scala:224)
[error] 	at sbt.Tests$$anonfun$7.apply(Tests.scala:196)
[error] 	at sbt.Tests$$anonfun$7.apply(Tests.scala:196)
[error] 	at sbt.std.Transform$$anon$3$$anonfun$apply$2.apply(System.scala:45)
[error] 	at sbt.std.Transform$$anon$3$$anonfun$apply$2.apply(System.scala:45)
[error] 	at sbt.std.Transform$$anon$4.work(System.scala:64)
[error] 	at sbt.Execute$$anonfun$submit$1$$anonfun$apply$1.apply(Execute.scala:237)
[error] 	at sbt.Execute$$anonfun$submit$1$$anonfun$apply$1.apply(Execute.scala:237)
[error] 	at sbt.ErrorHandling$.wideConvert(ErrorHandling.scala:18)
[error] 	at sbt.Execute.work(Execute.scala:244)
[error] 	at sbt.Execute$$anonfun$submit$1.apply(Execute.scala:237)
[error] 	at sbt.Execute$$anonfun$submit$1.apply(Execute.scala:237)
[error] 	at sbt.ConcurrentRestrictions$$anon$4$$anonfun$1.apply(ConcurrentRestrictions.scala:160)
[error] 	at sbt.CompletionService$$anon$2.call(CompletionService.scala:30)
[error] 	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
[error] 	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
[error] 	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
[error] 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
[error] 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
[error] 	at java.lang.Thread.run(Thread.java:745)
```

### PR DESCRIPTION
**Pull request by xuwei-k**
fix #671 (only MySQL and Postgres)

### CODE REVIEW

File: scalikejdbc-mapper-generator/src/sbt-test/scalikejdbc-mapper-generator/gen/src/test/scala/ProgrammersTest.scala [hunk 1]
```diff
@@ -11,7 +11,21 @@ class ProgrammersTest extends fixture.FlatSpec with Matchers with AutoRollback {
...
+    try {
+      props.load(inputStream)
+    } finally {
+      inputStream.close()
+    }
+    props.get("jdbc.driver").toString
+  }
+
   it should "be available" in { implicit session =>
+    if (driverClassName() == "org.h2.Driver") {
```
Comment (xuwei-k) [inline scalikejdbc-mapper-generator/src/sbt-test/scalikejdbc-mapper-generator/gen/src/test/scala/ProgrammersTest.scala]: I'm not sure why h2 failure 😢 

```
[error] scalikejdbc.InvalidColumnNameException: Invalid column name. (name: t_o__to, registered names: ID,NAME,T1,T2,T3,TYPE,TO_STRING,HASH_CODE,WAIT,GET_CLASS,NOTIFY,NOTIFY_ALL,PRODUCT_ARITY,PRODUCT_ITERATOR,PRODUCT_PREFIX)
[error] 	at scalikejdbc.SQLSyntaxSupportFeature$ColumnSQLSyntaxProvider.$anonfun$column$4(SQLSyntaxSupportFeature.scala:451)
[error] 	at scalikejdbc.SQLSyntaxSupportFeature$ColumnSQLSyntaxProvider$$Lambda$1773/1997680695.apply(Unknown Source)
[error] 	at scala.Option.getOrElse(Option.scala:121)
[error] 	at scalikejdbc.SQLSyntaxSupportFeature$ColumnSQLSyntaxProvider.$anonfun$column$1(SQLSyntaxSupportFeature.scala:450)
[error] 	at scalikejdbc.SQLSyntaxSupportFeature$ColumnSQLSyntaxProvider$$Lambda$1768/430272966.apply(Unknown Source)
[error] 	at scala.collection.concurrent.TrieMap.getOrElseUpdate(TrieMap.scala:903)
[error] 	at scalikejdbc.SQLSyntaxSupportFeature$ColumnSQLSyntaxProvider.column(SQLSyntaxSupportFeature.scala:449)
[error] 	at scalikejdbc.SQLSyntaxSupportFeature$SQLSyntaxProvider.c(SQLSyntaxSupportFeature.scala:309)
[error] 	at scalikejdbc.SQLSyntaxSupportFeature$SQLSyntaxProvider.c$(SQLSyntaxSupportFeature.scala:309)
[error] 	at scalikejdbc.SQLSyntaxSupportFeature$ColumnSQLSyntaxProvider.c(SQLSyntaxSupportFeature.scala:421)
[error] 	at scalikejdbc.SQLSyntaxSupportFeature$SQLSyntaxProvider.field(SQLSyntaxSupportFeature.scala:324)
[error] 	at scalikejdbc.SQLSyntaxSupportFeature$SQLSyntaxProvider.field$(SQLSyntaxSupportFeature.scala:319)
[error] 	at scalikejdbc.SQLSyntaxSupportFeature$ColumnSQLSyntaxProvider.field(SQLSyntaxSupportFeature.scala:421)
[error] 	at app.models.Programmers$.create(Programmers.scala:115)
[error] 	at app.models.ProgrammersTest.$anonfun$new$2(ProgrammersTest.scala:20)
[error] 	at app.models.ProgrammersTest$$Lambda$1760/2128850787.apply(Unknown Source)
[error] 	at scala.collection.TraversableLike.$anonfun$map$1(TraversableLike.scala:234)
[error] 	at scala.collection.TraversableLike$$Lambda$92/151412252.apply(Unknown Source)
[error] 	at scala.collection.immutable.List.foreach(List.scala:378)
[error] 	at scala.collection.TraversableLike.map(TraversableLike.scala:234)
[error] 	at scala.collection.TraversableLike.map$(TraversableLike.scala:227)
[error] 	at scala.collection.immutable.List.map(List.scala:284)
[error] 	at app.models.ProgrammersTest.$anonfun$new$1(ProgrammersTest.scala:19)
[error] 	at app.models.ProgrammersTest$$Lambda$1511/1305186059.apply(Unknown Source)
[error] 	at org.scalatest.fixture.TestSuite$TestFunAndConfigMap.$anonfun$apply$1(TestSuite.scala:132)
[error] 	at org.scalatest.fixture.TestSuite$TestFunAndConfigMap$$Lambda$1566/1521418053.apply(Unknown Source)
[error] 	at org.scalatest.OutcomeOf.outcomeOf(OutcomeOf.scala:85)
[error] 	at org.scalatest.OutcomeOf.outcomeOf$(OutcomeOf.scala:83)
[error] 	at org.scalatest.OutcomeOf$.outcomeOf(OutcomeOf.scala:104)
[error] 	at org.scalatest.fixture.TestSuite$TestFunAndConfigMap.apply(TestSuite.scala:132)
[error] 	at org.scalatest.fixture.TestSuite$OneArgTest$$anon$1.apply(TestSuite.scala:80)
[error] 	at org.scalatest.TestSuite.withFixture(TestSuite.scala:196)
[error] 	at org.scalatest.TestSuite.withFixture$(TestSuite.scala:195)
[error] 	at org.scalatest.fixture.FlatSpec.withFixture(FlatSpec.scala:226)
[error] 	at scalikejdbc.scalatest.AutoRollback.$anonfun$withFixture$1(AutoRollback.scala:63)
[error] 	at scalikejdbc.scalatest.AutoRollback$$Lambda$1564/228854960.apply(Unknown Source)
[error] 	at scalikejdbc.LoanPattern.using(LoanPattern.scala:18)
[error] 	at scalikejdbc.LoanPattern.using$(LoanPattern.scala:16)
[error] 	at app.models.ProgrammersTest.using(ProgrammersTest.scala:9)
[error] 	at scalikejdbc.scalatest.AutoRollback.withFixture(AutoRollback.scala:57)
[error] 	at scalikejdbc.scalatest.AutoRollback.withFixture$(AutoRollback.scala:56)
[error] 	at app.models.ProgrammersTest.withFixture(ProgrammersTest.scala:9)
[error] 	at org.scalatest.fixture.FlatSpecLike.invokeWithFixture$1(FlatSpecLike.scala:2131)
```

Comment (seratch) [inline scalikejdbc-mapper-generator/src/sbt-test/scalikejdbc-mapper-generator/gen/src/test/scala/ProgrammersTest.scala]: still failing? re-running the job fixed that?

Comment (xuwei-k) [inline scalikejdbc-mapper-generator/src/sbt-test/scalikejdbc-mapper-generator/gen/src/test/scala/ProgrammersTest.scala]: I've given up h2 test. still ProgrammersTest will fail only h2 if remove this `pending`

- https://github.com/xuwei-k/scalikejdbc/commit/ef3734220081da7e1b712e8837ba86d7e3c1266c
- https://travis-ci.org/xuwei-k/scalikejdbc/jobs/220021999

Comment (seratch) [inline scalikejdbc-mapper-generator/src/sbt-test/scalikejdbc-mapper-generator/gen/src/test/scala/ProgrammersTest.scala]: Going with this (pending for h2) is ok so far.

**Pull request merged**